### PR TITLE
Clean up hydroelastic-contact tutorial

### PR DESCRIPTION
**Pull request by DamrongGuoy**
- discrete_contact_approximation="similar" for high impact
  - Remove simulator.Initialize()
  - Save simulation result into html file

- - -
This change is [Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)

### CODE REVIEW

**Review by amcastro-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 1 unresolved discussion, needs platform reviewer assigned, needs at least two assigned reviewers, labeled "do not merge" (waiting on @DamrongGuoy)

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 435 at r2](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NlsZ83M5IIdpVqbe6HU:-NlsZ83M5IIdpVqbe6HV:b-gqknzo) ([raw file](https://github.com/RobotLocomotion/drake/blob/5ab2e72a7582021c0b39bbaba8caf05930edf495/tutorials/hydroelastic_contact_basics.ipynb#L435)):*
> ```
>     "          <drake:mu_dynamic>0.5</drake:mu_dynamic>\n",
>     "          <drake:hunt_crossley_dissipation>1.25</drake:hunt_crossley_dissipation>\n",
>     "          <drake:relaxation_time>0.1</drake:relaxation_time>\n",
> ```

if you remove the `relaxation_time` and choose "sap", you'll be using the default relaxation time. Is that what you want?
I believe what you wan here is to change the approximation to "lagged" and probably even leave the `relaxation_time` in there for those wanting to play with the "sap" approximation, it doesn't hurt and it might have pedagogical value.

For more info, read the docs for `DiscreteContactApproximation` (enum in `multibody_plant.h`

**Review by DamrongGuoy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 1 unresolved discussion, needs platform reviewer assigned, needs at least two assigned reviewers, labeled "do not merge" (waiting on @DamrongGuoy)

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 435 at r2](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NlsZ83M5IIdpVqbe6HU:-NlyTEZn5RbsFZPBlSNY:b-glhdp8) ([raw file](https://github.com/RobotLocomotion/drake/blob/5ab2e72a7582021c0b39bbaba8caf05930edf495/tutorials/hydroelastic_contact_basics.ipynb#L435)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

if you remove the `relaxation_time` and choose "sap", you'll be using the default relaxation time. Is that what you want?
I believe what you wan here is to change the approximation to "lagged" and probably even leave the `relaxation_time` in there for those wanting to play with the "sap" approximation, it doesn't hurt and it might have pedagogical value.

For more info, read the docs for `DiscreteContactApproximation` (enum in `multibody_plant.h`
</blockquote></details>

Thank you.  That's very helpful.  I'll read [DiscreteContactApproximation](https://drake.mit.edu/doxygen_cxx/namespacedrake_1_1multibody.html#aa1d02176591187cb83e77e9ff779d679).

Set to Working.  Could you remove the *No Entry* ⛔, please?

**Review by amcastro-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 1 unresolved discussion, needs platform reviewer assigned, needs at least two assigned reviewers, labeled "do not merge" (waiting on @DamrongGuoy)

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 435 at r2](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NlsZ83M5IIdpVqbe6HU:-Nm2dgtZ63h3D7MIldoX:b1gp1pk) ([raw file](https://github.com/RobotLocomotion/drake/blob/5ab2e72a7582021c0b39bbaba8caf05930edf495/tutorials/hydroelastic_contact_basics.ipynb#L435)):*
<details><summary><i>Previously, DamrongGuoy (Damrong Guoy) wrote…</i></summary><blockquote>

Thank you.  That's very helpful.  I'll read [DiscreteContactApproximation](https://drake.mit.edu/doxygen_cxx/namespacedrake_1_1multibody.html#aa1d02176591187cb83e77e9ff779d679).

Set to Working.  Could you remove the *No Entry* ⛔, please?
</blockquote></details>

Good!
yes, I believe I just did. Otherwise feel free to dismiss me from any blocking conversation. I'll be traveling starting tomorrow till January 14.

**Review by DamrongGuoy — COMMENTED** [review]
-(status: do not merge) -(status: do not review) 

+@rpoyner-tri for feature review, please.

Just a reminder that you can run the tutorial locally by:
```
bazel run //tutorials:hydroelastic_contact_basics
```

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-:-NnHSzIG0-JoddVyY_YI:bdg9kl6)* status: LGTM missing from assignee rpoyner-tri(platform), needs at least two assigned reviewers, labeled "do not merge" (waiting on @DamrongGuoy)

**Review by rpoyner-tri — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-:-NnKeEVo3dqAoNwnnnw_:bnfp4nl)* status: 1 unresolved discussion, needs at least two assigned reviewers

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 796 at r3](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NnKU2Q3BopTYo6oMjEj:-NnKU2Q3BopTYo6oMjEk:brpq8m) ([raw file](https://github.com/RobotLocomotion/drake/blob/b948b064ed4e18912b68c0b588b969464e5a85ec/tutorials/hydroelastic_contact_basics.ipynb#L796)):*
> ```
>     "In the next code block, you will run the simulation for 6 seconds to steady state, and you should see `f.Z()` around 9.81 newtons equal to the gravity force:\n",
>     "\n",
>     ">   f=[-3.3583690999835774e-05, -4.041996578714733e-06, 9.809999999745465]"
> ```

I doubt all of these digits are significant, or stable (say, for example, on macos). Can we trim them down, similar to the cell above?

**Review by DamrongGuoy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 1 unresolved discussion, needs at least two assigned reviewers

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 796 at r3](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NnKU2Q3BopTYo6oMjEj:-Nne_1nU2A-deRdsQE3v:b-i8bzil) ([raw file](https://github.com/RobotLocomotion/drake/blob/b948b064ed4e18912b68c0b588b969464e5a85ec/tutorials/hydroelastic_contact_basics.ipynb#L796)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

I doubt all of these digits are significant, or stable (say, for example, on macos). Can we trim them down, similar to the cell above?
</blockquote></details>

Done. Thanks.

I'd love to run it on MacOS and compare.  Does our equipment support that? (For example, build and run on MacSim, and show the notebook on Puget?).

**Review by DamrongGuoy — COMMENTED** [review]
+@SeanCurtis-TRI for platform review, please.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-:-Nne_nv4AOFXtdNFO_eU:b-6cxuoj)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform)

**Review by SeanCurtis-TRI — COMMENTED** [review]
:LGTM:

> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-:-NnetB2L2Sy_eOe01gs3:bmxilld)* status: 3 unresolved discussions

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 926 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NneeUMU5q7LY2xCw_z3:-NneeUMU5q7LY2xCw_z4:btun7za) ([raw file](https://github.com/RobotLocomotion/drake/blob/62a883c5dfd60b639ee65a24fee5877c70d1ca81/tutorials/hydroelastic_contact_basics.ipynb#L926)):*
> ```
>     "### Save simulation result into a html file for sharing\n",
>     "\n",
>     "You can download the simulation result into a html file. You can share the html file with your friends for playback without the source code. Remove `#` below and run the code. You should get a html file that you can open in a web browser on any computer."
> ```

BTW "without the source code" undersells the benefit. You don't need the binary or even to *run* the simulation.



_Suggestion:_
```
You can download the simulation result into a self-contained html file, allowing others to playback the simulated results without simulating.
```
___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 926 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-Nnest1a7yrR45qrkfep:-Nnest1a7yrR45qrkfeq:bclj11c) ([raw file](https://github.com/RobotLocomotion/drake/blob/62a883c5dfd60b639ee65a24fee5877c70d1ca81/tutorials/hydroelastic_contact_basics.ipynb#L926)):*
> ```
>     "### Save simulation result into a html file for sharing\n",
>     "\n",
>     "You can download the simulation result into a html file. You can share the html file with your friends for playback without the source code. Remove `#` below and run the code. You should get a html file that you can open in a web browser on any computer."
> ```

What's the motivation for having the code commented out and asking the user to uncomment them?

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 4 unresolved discussions

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 936 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NnetrDT6VeuXO5UeIqM:-NnetrDT6VeuXO5UeIqN:b7aypv) ([raw file](https://github.com/RobotLocomotion/drake/blob/62a883c5dfd60b639ee65a24fee5877c70d1ca81/tutorials/hydroelastic_contact_basics.ipynb#L936)):*
> ```
>    "source": [
>     "# import webbrowser\n",
>     "# webbrowser.open(f\"{meshcat.web_url()}/download\")"
> ```

Don't abuse `webbrower` just to download the HTML.  If you want the notebook code to "download" the HTML, then call `meshcat.StaticHtml()` to get the HTML as a `str`, and then separately write that string to a temp filename.

Or if you like, you can tell people about the URL trick in the prose here, or even print the URL to the log so that they can click it to start the download.

Bottom line: `webbrowser` is appropriate in code only when we want to open a new tab for the user to interact with.

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 3 unresolved discussions

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 796 at r3](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NnKU2Q3BopTYo6oMjEj:-Nnj6-KO3f8apKF3zdZs:bos9q2y) ([raw file](https://github.com/RobotLocomotion/drake/blob/b948b064ed4e18912b68c0b588b969464e5a85ec/tutorials/hydroelastic_contact_basics.ipynb#L796)):*
<details><summary><i>Previously, DamrongGuoy (Damrong Guoy) wrote…</i></summary><blockquote>

Done. Thanks.

I'd love to run it on MacOS and compare.  Does our equipment support that? (For example, build and run on MacSim, and show the notebook on Puget?).
</blockquote></details>

  You can. Just ssh with ` LocalForward 8888 localhost:8888` in the `~/.ssh/config` stanza for  the macsim machine. I did, and got:
```
 f=\[-3.358369654267214e-05, -4.042006364273084e-06, 9.809999999746701\],
```

**Review by rpoyner-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 3 unresolved discussions

**Review by DamrongGuoy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 2 unresolved discussions

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 796 at r3](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NnKU2Q3BopTYo6oMjEj:-NnkuacDCQ1EOREJXqsm:b-9755fn) ([raw file](https://github.com/RobotLocomotion/drake/blob/b948b064ed4e18912b68c0b588b969464e5a85ec/tutorials/hydroelastic_contact_basics.ipynb#L796)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

You can. Just ssh with ` LocalForward 8888 localhost:8888` in the `~/.ssh/config` stanza for  the macsim machine. I did, and got:
```
 f=\[-3.358369654267214e-05, -4.042006364273084e-06, 9.809999999746701\],
```
</blockquote></details>

Thank you!
___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 926 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-Nnest1a7yrR45qrkfep:-Nnkum7y1e9RAhZADrO_:b-7qhjf0) ([raw file](https://github.com/RobotLocomotion/drake/blob/62a883c5dfd60b639ee65a24fee5877c70d1ca81/tutorials/hydroelastic_contact_basics.ipynb#L926)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

What's the motivation for having the code commented out and asking the user to uncomment them?
</blockquote></details>

Done. Uncomment now.

Previously if I didn't comment it out, it would download and renamed files to `meshcat.html`, `meshcat (1).html`, `meshcat (2).html`,... every time users re-run the notebook.

Now I only print the download URL without calling `webbrowser`, so users can decide whether to download by clicking on the URL.
___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 936 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NnetrDT6VeuXO5UeIqM:-NnkurL96mXTKyWCo9wR:b5johdb) ([raw file](https://github.com/RobotLocomotion/drake/blob/62a883c5dfd60b639ee65a24fee5877c70d1ca81/tutorials/hydroelastic_contact_basics.ipynb#L936)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Don't abuse `webbrower` just to download the HTML.  If you want the notebook code to "download" the HTML, then call `meshcat.StaticHtml()` to get the HTML as a `str`, and then separately write that string to a temp filename.

Or if you like, you can tell people about the URL trick in the prose here, or even print the URL to the log so that they can click it to start the download.

Bottom line: `webbrowser` is appropriate in code only when we want to open a new tab for the user to interact with.
</blockquote></details>

Done.  Thanks.  I print the URL and tell people to download it if they want to.  No more `webbrowser`.

**Review by DamrongGuoy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: 1 unresolved discussion

___
*[`tutorials/hydroelastic_contact_basics.ipynb` line 926 at r4](https://reviewable.io/reviews/RobotLocomotion/drake/20665#-NneeUMU5q7LY2xCw_z3:-NnkxfEJ-EQmxMne4BOC:boyexbg) ([raw file](https://github.com/RobotLocomotion/drake/blob/62a883c5dfd60b639ee65a24fee5877c70d1ca81/tutorials/hydroelastic_contact_basics.ipynb#L926)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW "without the source code" undersells the benefit. You don't need the binary or even to *run* the simulation.
</blockquote></details>

Done. Thanks.

**Review by SeanCurtis-TRI — COMMENTED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20665)* status: <span>:shipit:</span> complete! all discussions resolved, LGTM from assignees rpoyner-tri(platform),SeanCurtis-TRI(platform)
